### ISSUE TITLE
.NET Core 2.1 WinHttpHandler proxy authentication broken when using PAC files

### ISSUE THREAD

**Issue by davidsh** [dotnet/corefx#30191]
This issue split off from #30166.

Proxy authentication using system default proxy settings that involve a PAC file (either autodiscovery or explicit PAC file) cause a NullReferenceException in WinHttpHandler.

![image](https://user-images.githubusercontent.com/7684835/41113984-f179e8ac-6a37-11e8-8d35-022c55383431.png)

This is a regression from .NET Core 2.0 caused by PR #28105

**Comment by davidsh** [comment]
Unlike #30166 which is specific to a proxy server using Windows auth schemes (Negotiate, NTLM), this issue occurs for any auth scheme (Negotiate, NTLM, Digest, Basic) that a proxy server might be using.

**Comment by davidsh** [comment]
Currently testing this is difficult in CI since it requires machine configuration, hosting a PAC file etc.  But these are the manual repro steps.

1. Configure network to use a PAC file for proxy settings. It can be either "Automatically detect settings" or "Use automatic configuration script".  The former requires configuring the network for WPAD protocol. The latter simply requires hosting a PAC file on an explicit server path, i.e.

![image](https://user-images.githubusercontent.com/7684835/41114198-922f5dcc-6a38-11e8-81c8-307e090f5800.png)

2. Have the contents of the PAC file point to an authenticating proxy server.  I.e.
```Javascript
function FindProxyForURL(url, host) { return "PROXY [::1]:3128"; }
```

3. Use WinHttpHandler either directly or via HttpClientHandler() constructor and using AppContext switch to turn off SocketsHttpHandler.

```c#
var handler = new HttpClientHandler();

// This line is not really needed since the NullReferenceException occurs before any credential
// would be used.
handler.DefaultProxyCredentials = new NetworkCredential("username", "password", "domain");

var client = new HttpClient(handler);

// NullReferenceException occurs calling GetAsync()
HttpResponseMessage response = await client.GetAsync("http://someserver");

```

**Comment by davidsh** [comment]
Re-open to target release/2.1.x servicing release.

**Comment by davidsh** [comment]
Fixed in release/2.1 with #30212
